### DISCUSSION TITLE
zip arrays

### DISCUSSION THREAD

**Discussion by trose-neuro** [bonsai-rx/bonsai#1528] [Q&A] [answered]
Hi all!

This is probably a surprisingly uninformed question - but how do I zip three three-column managed arrays (float[]) so the zipped output can be written as nine-column csv? 

![image](https://github.com/bonsai-rx/bonsai/assets/4570253/a1ecfc89-28fd-4cf5-9afc-dc7b3c476b38)

CSV output is: `3,3,1,False,True,False,3,3,1,False,True,False,3,3,1,False,True,False`

and not the joined sensor data. Minimal workflow:

[WearCSV_git.bonsai.zip](https://github.com/bonsai-rx/bonsai/files/12388366/WearCSV_git.bonsai.zip)

I am trying to convert the Matrix output from the HARP WEAR events to float first and then write all values to a joint csv. This became necessary after the 2.8 update which seems to have broken the joint output of all sensors in the "Wear.Motion" event type:

![image](https://github.com/bonsai-rx/bonsai/assets/4570253/ddf9adaa-8a03-4b37-bd14-ce044766fc6a)

Is this the beginning of the HARP WEAR deprecation? 

Thanks a lot for your help!

Tobias

**Comment by bruno-f-cruz** [comment, marked as answer]
Arrays should be saved as a flat binary file using matrixwriter. I would advise you to save them to separate files if the data has different formats (ie double vs byte). If you do want to save them to the same file, you must first convert them to the same data format and concatenate the arrays.

If you want to convert the arrays to its single elements you can use a Merge(rx) after an array.

**Reply by trose-neuro** [reply]
Thanks! I did have some trouble parsing the full "Wear.Motion" event binary from MatrixWriter before and therefore went for csv. Merge(rx) works well - but I might indeed go binary for the separate events again. You were right (discord) that an update to the HARP libraries was the most likely culprit. I was unable, however, to restore the previous state (Bonsai.config - environments for the future). It works well for us like this, though.
